### PR TITLE
Add pairing logic and ts-node

### PR DESCRIPTION
**Pull request by VKong6019**
hey buddies,

I saw jia start the slack bot dms 👀, so I thought I just make this into a pr to expedite the process. 

The way the pairing works in `generatePair.ts` is that it'll partition an array of users in a channel by 2s. If there's an uneven number of people, then add that loner into some existing pair. 

Idk if this is added in my commits, but I installed `ts-node`  and used that to run typescript files easily.
so then I just do `ts-node pairAndIntroduce.ts`
https://www.npmjs.com/package/ts-node

### CODE REVIEW

File: utils/generatePairs.ts [hunk 1]
```diff
@@ -1,12 +1,29 @@
 import { App } from "@slack/bolt";
 
 const TESTING_CHANNEL_ID = "C02J6R0SUSX";
```
Comment (VKong6019) [inline utils/generatePairs.ts:3]: note: will need to change this later

File: utils/generatePairs.ts [hunk 2]
```diff
@@ -1,12 +1,29 @@
...
+    //copy pasta internet go brrr
+    for (let i = 0; i < memberIDs.length; i += 2) {
+        const chunk = memberIDs.slice(i, i + 2);
+        pairs.push(chunk);
+    }
+
+    // only works if we're doing pairs
+    // if someone is lonely, then group that person with an existing pair
+    if (pairs.length > 1 && pairs.length * 2 !== memberIDs.length) {
+        const lonelyMember: string[] = pairs.pop() ?? [];
```
[the remark this answers is not in the document]
Comment (ryandrew14) [reply utils/generatePairs.ts:20]: this thing so useful i swear its like every 3 lines I write

File: utils/generatePairs.ts (hunk 1 above)
Comment (ryandrew14) [reply utils/generatePairs.ts:3]: could u write todo comment

**Review by ryandrew14 — CHANGES_REQUESTED** [review]
looks great tysm but it's actually good practice to commit package-lock.json, see here: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

also any chance u could write tests for this one 😄 lol it would be a nice easy thing to set up Jest with and since ur the test wizard of the team

File: __tests__/generatePairs.test.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        mockApp.client.conversations = mockConversations;
+
+        const actualList: string[][] = await generatePairs(mockApp);
+
+        expect(actualList).toStrictEqual([]);
+        expect(mockConversations.members).toHaveBeenCalled();
+    })
+
+    // this would never happen hopefully
+    it('should pair a list with 1 member', async () => {
```
Comment (ryandrew14) [inline __tests__/generatePairs.test.ts:51]: typo

**Review by ryandrew14 — APPROVED** [review]
🤠 yeehaw

**Pull request merged**